src/lib/mina_lib/mina_lib.ml

```diff
@@ -917,31 +917,21 @@ let add_complete_work ~logger ~fee ~prover
   |> Result.iter_error ~f:(fun err ->
          [%log warn] "Failed to update metrics on adding work"
            ~metadata:[ ("error", `String (Error.to_string_hum err)) ] ) ;
-  let cb result =
-    (* remove it from seen jobs after attempting to adding it to the pool to
-           avoid this work being reassigned.
-     * If the diff is accepted then remove it from the seen jobs.
-     * If not then the work should have already been in the pool with a 
-           lower fee or the statement isn't referenced anymore or any other 
-           error. In any case remove it from the seen jobs so that it can be 
-           picked up if needed *)
-    Work_selector.remove t.work_selector stmts ;
```

Comment:
If diff is not accepted, we don't need it as it's sad here it's replaced by a lower fee work by some other worker's proof; 
If it's accepted we don't need to remove as well -- the selector now doesn't regenerate the samething anyway, once scheduled it's marked as scheduled and reseting won't happen.